clarity/utils/audiogram.py

```diff
@@ -173,3 +175,73 @@ def resample(self, new_frequencies: ndarray) -> Audiogram:
     frequencies=FULL_STANDARD_AUDIOGRAM_FREQUENCIES,
     levels=np.array([19, 19, 28, 35, 40, 47, 52, 56, 58, 58, 63, 70, 75, 80, 80]),
 )
+
+
+@dataclass
+class Listener:
+    """Dataclass to represent a Listener.
+
+    The listener is currently defined by their left and right ear
+    audiogram. In later versions, this may be extended to include
+    further audiometric data.
+
+    The class provides methods for reading metadata files which
+    will also include some basic validation.
+
+    Attributes:
+        id (str): The ID of the listener
+        audiogram_left (Audiogram): The audiogram for the left ear
+        audiogram_right (Audiogram): The audiogram for the right ear
+    """
+
+    id: str
+    audiogram_left: Audiogram
+    audiogram_right: Audiogram
+
+    @staticmethod
+    def from_dict(listener_dict: dict) -> Listener:
+        """Create a Listener from a dict.
+
+        The dict structure and fields are based on those used
+        in the Clarity metadata files.
+
+        Args:
+            listener_dict (dict): The listener dict
+
+        Returns:
+            Listener: The listener
+
+        """
+        return Listener(
+            id=listener_dict["name"],
+            audiogram_left=Audiogram(
+                levels=listener_dict["audiogram_levels_l"],
+                frequencies=listener_dict["audiogram_cfs"],
+            ),
+            audiogram_right=Audiogram(
+                levels=listener_dict["audiogram_levels_r"],
+                frequencies=listener_dict["audiogram_cfs"],
+            ),
+        )
+
+    @staticmethod
+    def read_listener_dict(filename: Path) -> dict[str, Listener]:
```

Comment:
An a bit pedantic comment but, maybe called as `read_listener_json` or `read_listener_file` as `read_listener_dict` may be misleading as `dict` is a data structure not a file format.

Reply:
Thanks.  I will give this some more thought after integrating into the recipes.  Perhaps read_listener_file is most consistent. The problem here is that in clarity nearly all the metadata has been distributed as a list of dicts -- which is really standard, and reading the files returns a list as expected... but the listeners were for some reason saved in a dictionary structure. So I wanted to highlight that the function returned a dictionary structure, ie. 

```
listeners = read_listener_dict("listeners.json")
```

It might even be best then if this function just flattened the dictionary into a list of values.

Reply:
Renamed as "load_listener_dict" which is perhaps more intuitive.